src/webgpu/util/f32_interval.ts

```diff
@@ -315,6 +369,53 @@ function roundAndFlushTernaryToInterval(
   return F32Interval.span(...intervals);
 }
 
+/** Converts a pair of vectors to an acceptance interval, using a specific function
+ *
+ * This handles correctly rounding and flushing inputs as needed.
+ * Duplicate inputs are pruned before invoking op.impl.
+ * All unique combinations of x & y are run.
+ * op.extrema is invoked before this point in the call stack.
+ * op.domain is tested before this point in the call stack.
+ *
+ * @param x first param to flush & round then invoke op.impl on
+ * @param y second param to flush & round then invoke op.impl on
+ * @param op operation defining the function being run
+ * @returns a span over all of the outputs of op.impl
+ */
+function roundAndFlushVectorPairToInterval(
+  x: number[],
+  y: number[],
+  op: VectorPairToIntervalOp
+): F32Interval {
+  assert(
+    x.every(e => !Number.isNaN(e)),
```

Comment:
.every and .some are much much better then trying to implement this correctly using .filter and .length